### PR TITLE
Add documentation for "NULL" cipher and digest algorithms.

### LINKED ISSUE #20340: The default provider documentation does not list the NULL cipher
This page is supposed to list the ciphers that are available in the default provider:

https://www.openssl.org/docs/man3.0/man7/OSSL_PROVIDER-default.html

Under the "Symmetric ciphers" section it should list the NULL cipher - but it does not. Similarly there is no "EVP_CIPHER-NULL" page to give an overview for the cipher, e.g. compare to "EVP_CIPHER-AES" here:

https://www.openssl.org/docs/man3.0/man7/EVP_CIPHER-AES.html

### PR DESCRIPTION
**Pull request by slontis**
Fixes #20340

##### Checklist

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: doc/man7/EVP_CIPHER-NULL.pod [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+RFC 5246 section-6.2.3.1
+
+=head1 SEE ALSO
+
+L<provider-cipher(7)>, L<OSSL_PROVIDER-default(7)>
+
+=head1 HISTORY
+
+The NULL cipher was added in OpenSSL version 3.0.
```
Comment (paulidale) [inline doc/man7/EVP_CIPHER-NULL.pod:65]: I'm pretty sure it predates 3.0.  I suggest just leaving the history section out.

File: doc/man7/EVP_MD-NULL.pod [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+in L<EVP_MD-common(7)>.
+
+=head1 SEE ALSO
+
+L<EVP_MD_CTX_set_params(3)>, L<provider-digest(7)>,
+L<OSSL_PROVIDER-default(7)>
+
+=head1 HISTORY
+
+The NULL digest was added in OpenSSL version 3.0.
```
Comment (paulidale) [inline doc/man7/EVP_MD-NULL.pod:35]: Ditto

File: doc/man7/EVP_CIPHER-NULL.pod (hunk 1 above)
Comment (mattcaswell) [reply doc/man7/EVP_CIPHER-NULL.pod:65]: At least in 1.1.1 there is an `EVP_enc_null()`

File: doc/man7/EVP_MD-NULL.pod (hunk 2 above)
Comment (mattcaswell) [reply doc/man7/EVP_MD-NULL.pod:35]: Similarly in 1.1.1 there is an `EVP_md_null()`

**Review by mattcaswell — CHANGES_REQUESTED** [review]

**Comment by slontis** [comment]
> At least in 1.1.1 there is an EVP_enc_null()

This refers to EVP_CIPHER_NULL.

Would 'The provider NULL cipher was ...' be sufficient here?

**Comment by slontis** [comment]
I dont think we should be leaving history sections out anywhere.

**Comment by paulidale** [comment]
> Would 'The provider NULL cipher was ...' be sufficient here?

This would be okay if the same is added for all other algorithms.  Having history for just these two when we don't **everywhere** else is wrong.

**Comment by slontis** [comment]
You can end up with 20 year old files with no history if you take the 'lets not add history approach'

**Comment by paulidale** [comment]
As noted: add the history globally.

**Review by paulidale — APPROVED** [review]

**Review by mattcaswell — APPROVED** [review]

**Comment by paulidale** [comment]
Merged to all three branches.

**Pull request closed without merging**